### PR TITLE
Impl get token in tycho client

### PR DESCRIPTION
**Pull request by zizou0x**

### CODE REVIEW

File: tycho-client/src/rpc.rs [hunk 1]
```diff
@@ -474,4 +526,77 @@ mod tests {
...
+                symbol: "USDC".to_string(),
+                decimals: 6,
+                tax: 0,
+                gas: vec![Some(40652)],
+            },
+        ];
+
+        mocked_server.assert();
+        assert_eq!(response.tokens, expected);
+        assert_eq!(response.pagination, PaginationParams { page: 0, page_size: 2 });
```
Comment (louise-poole) [inline tycho-client/src/rpc.rs:639]: Should this not be page 1? From my understanding the response should hold the next page that you will want to request.

Comment (louise-poole) [reply tycho-client/src/rpc.rs:639]: Ah, it's a hardcoded test asset 😆 AG then

File: tycho-client/src/rpc.rs [hunk 2]
```diff
@@ -287,11 +293,57 @@ impl RPCClient for HttpRPCClient {
 
         Ok(states)
     }
+
+    async fn get_tokens(
```
Comment (tvinagre) [inline tycho-client/src/rpc.rs:339]: I think you'll need a `get_all_tokens`  as this function will hardly be used solo. 
`get_all_tokens` should spawn the request with different pages until it gets an empty response. 
Also, would be nice to document this function to be explicit that this just spawns the request and parses the response, but doesn't handle pagination

File: tycho-client/src/rpc.rs (hunk 1 above)
Comment (tvinagre) [reply tycho-client/src/rpc.rs:639]: No, the response is just a copy from the request actually. If it had the next page it'd be smth like a cursor

**Pull request merged**